src/math/log.c

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/log.h>
+
+/**
+ *  Base-2 logarithm
+ *
+ * Y = (u) computes the base-2 logarithm of
+ * u using lookup table.
+ * input u must be scalar/real number and positive
+ * Arguments	: uint32_t u[Q31.1]
```

Comment:
So "u" is a positive integer? Q31.1 is confusing because it is not Q-formatted and it is unsigned. But ".1" means there is one fraction bit?

Reply:
yes `u` is positive. Your interpretation / understanding is right . `Word length` is `31` bit with `1 fractional bit length` and no `signed` bit `0` . I will add bit details in function header description

Reply:
I am curious why choosing 1 fractional bit?

Also I would recommend to provide log10() and log() (natural logarithm) functions with proper precision in Q-format with the leverage of log2() you implemented. Maybe in another file.
From my perspective log10() and log() are used more frequently than log2().

Reply:
_I am curious why choosing 1 fractional bit?_

2^31 is 2147483647.5 with 1 fractional bit. Look at the testvector in `log2_tables.h` last element 4294967295U which 4294967295U/2 = 2147483647.5. This means `log2(2147483647.5) = 30.9999999996641 ` i.e. full 31 bit range of input data is explored. I will add table having range and Q format information

`_provide log10() and log() (natural logarithm) functions_`
log10() and log() are derivative of log2().   yes, I can do in a separate PR and files.